container-storage-setup.sh

```diff
@@ -755,26 +756,33 @@ create_partition() {
     Fatal "udevadm settle after partition creation failed. Exiting."
   fi
 
-  if ! wait_for_dev ${dev}1; then
-    Fatal "Partition device ${dev}1 is not available"
+  part=$(query_first_child $dev)
+
+  if ! wait_for_dev ${part}; then
+    Fatal "Partition device ${part} is not available"
   fi
+  echo $part
+}
+
+query_first_child() {
+  lsblk -npl -o NAME "$1" | tail -n +2 | head -1
 }
 
 create_disk_partitions() {
-  local devs="$1"
+  local devs="$1" part
 
   for dev in $devs; do
-    create_partition $dev
+    part=$(create_partition $dev)
```

Comment:
how about querying partition after creation. Instead of returning it from create_partition. 

Anyway, create_partition() itself does not know better as it also is querying the partition name after creating it.

Downside is that we will query twice. But it removes the need of capturing the stdout of create_partition() and need of supressing sfdisk output. I like that better.